src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -399,6 +399,11 @@ private Task ProcessGoAwayFrameAsync()
                 throw new Http2ConnectionErrorException(Http2ErrorCode.PROTOCOL_ERROR);
             }
 
+            if (_incomingFrame.StreamId != 0)
+            {
+                throw new Http2ConnectionErrorException(Http2ErrorCode.PROTOCOL_ERROR);
```

Comment:
For our logging purposes we should include messages describing these errors.

Reply:
Yes, I will revisit that (#2080).